### ISSUE TITLE
Errors disappear when thrown within generator function

### ISSUE THREAD

**Issue by bjb568** [nemisj/yield-yield#5]
require('yield-yield')(function*() { throw new Error(); })()

I would expect this would throw an error, just like this:

    (function() { throw new Error(); })()

But the error in the first case just disappears and causes anything remaining in the function to not be executed. This makes debugging harder since you can't actually see the error.

**Comment by nemisj** [comment]
It's because all the errors are catched and passed to the callback as first argument.

normally you would call it like this:

```
require('yield-yield')(function*() { throw new Error(); })(function (err) {
    console.log('err', err);
})
```

It's also the way all async stuff works in node.js. since this lib is a helper and must be used in conjunction with the normal async code, it's working this way.

**Comment by bjb568** [comment]
It would make sense for it to default to throwing the error. Blindly catching errors by default can lead to inconsistent application state which is worse than a crash which can be handled by forever or its equivalents.

Many functions in node, including built-in ones like http.createServer, call a function without specifying a callback. If the function happens to be created with yield-yield, errors thrown in it should still cause a crash rather than just be sent into the void.

**Comment by nemisj** [comment]
make sense, i will see how to throw errors if no callback is specified

**Comment by nemisj** [comment]
i've published version v1.1.0 which has this functionality
